### PR TITLE
Use new analyzer/fixer in roslyn

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: eng/targets/Imports.targets [hunk 1]
```diff
@@ -45,7 +45,7 @@
       impact the correctness of the output, so we avoid the performance overhead where it's easy to do so.
     -->
     <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND '$(DisableNullableWarnings)' == 'true'">false</RoslynCheckCodeStyle>
-    <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND ('$(ContinuousIntegrationBuild)' != 'true' OR '$(RoslynEnforceCodeStyle)' == 'true')">true</RoslynCheckCodeStyle>
+    <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND ('$(ContinuousIntegrationBuild)' != 'true' OR '$(RoslynEnforceCodeStyle)' == 'true')">false</RoslynCheckCodeStyle>
```
Comment (CyrusNajmabadi) [inline eng/targets/Imports.targets:48]:
```suggestion
    <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND ('$(ContinuousIntegrationBuild)' != 'true' OR '$(RoslynEnforceCodeStyle)' == 'true')">true</RoslynCheckCodeStyle>
```

File: src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs [hunk 2]
```diff
@@ -608,12 +605,9 @@ private void CreateEvaluationResultAndContinue(EvalResult result, WorkList workL
                     formatSpecifiers,
                     flags,
                     Formatter2.GetEditableValueString(value, inspectionContext, declaredTypeAndInfo.Info));
-                if (expansion == null)
-                {
-                    expansion = value.HasExceptionThrown()
+                expansion ??= value.HasExceptionThrown()
```
Comment (akhera99) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs:608]: I think in cases like this I would rather be verbose

**Review by akhera99 — APPROVED** [review]
Looks cool!

File: src/VisualStudio/Core/Def/Interactive/ScriptingOleCommandTarget.cs [hunk 3]
```diff
@@ -29,10 +29,7 @@ internal sealed class ScriptingOleCommandTarget : AbstractOleCommandTarget
         {
             var result = WpfTextView.GetBufferContainingCaret(contentType: ContentTypeNames.RoslynContentType);
 
-            if (result == null)
-            {
-                result = WpfTextView.GetBufferContainingCaret(contentType: PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
-            }
+            result ??= WpfTextView.GetBufferContainingCaret(contentType: PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
```
Comment (JoeRobich) [inline src/VisualStudio/Core/Def/Interactive/ScriptingOleCommandTarget.cs:32]: For this scenario I would prefer something like

```csharp
return WpfTextView.GetBufferContainingCaret(contentType: ContentTypeNames.RoslynContentType)
    ?? WpfTextView.GetBufferContainingCaret(contentType: PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
```

File: src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrRuntimeInstance.cs [hunk 4]
```diff
@@ -47,10 +47,7 @@ internal DkmClrRuntimeInstance(Assembly assembly) : this(new[] { assembly })
             bool enableNativeDebugging = false)
             : base(enableNativeDebugging)
         {
-            if (getModule == null)
-            {
-                getModule = (r, a) => new DkmClrModuleInstance(r, a, (a != null) ? new DkmModule(a.GetName().Name + ".dll") : null);
-            }
+            getModule ??= (r, a) => new DkmClrModuleInstance(r, a, (a != null) ? new DkmModule(a.GetName().Name + ".dll") : null);
```
Comment (davidwengier) [inline src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrRuntimeInstance.cs:50]: ahh the old null coalescing lambda with a ternary.. if i had a dollar for every time i've seen that!

Comment (CyrusNajmabadi) [reply src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrRuntimeInstance.cs:50]: tis indeed glorious.  bask in it and feel the warmth shine on your face.

File: src/VisualStudio/Core/Def/Interactive/ScriptingOleCommandTarget.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Interactive/ScriptingOleCommandTarget.cs:32]: yup!  that seems like a nice additional improvement we could do in this space.  e.g. "prefer coalesce vs multiple assignments and tests"

**Review by davidwengier — APPROVED** [review]

**Pull request merged**